src/App.js

```diff
@@ -6,41 +6,136 @@ function App() {
   const [todoList, setTodoList] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
 
-  useEffect(() => {
-    new Promise((resolve, reject) =>
-      setTimeout(
-        () =>
-          resolve({
-            data: {
-              todoList: JSON.parse(localStorage.getItem("savedTodoList")) || [],
-            },
-          }),
-        2000
-      )
-    ).then((result) => {
-      setTodoList(result.data.todoList);
+// Create a new variable url
+const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}`;
+
+// New async function fetchData
+  const fetchData = async () => {
+    // Declare empty object variable options
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
+    };
+
+    //console.log('API Token:', process.env.REACT_APP_AIRTABLE_API_TOKEN);
+    //console.log('API Token:', process.env.REACT_APP_AIRTABLE_BASE_ID);
+
+    try {
+      // Fetch data from Airtable
+      const response = await fetch(url, options);
+
+      // Check if the response is okay, throw an error if not
+      if (!response.ok) {
+        // throw new Error(`Error: ${response.status}`);
+       // or 2nd option
+        const message = `Error: ${response.status}`;
+        throw new Error(message);
+      }
+
+      // Parse the response data
+      const data = await response.json();
+
+      // Console.log the data for observation
+      console.log('example response:', data);
+
+      // Map data.records into an array of todo objects
+      const todos = data.records.map((todo) => {
+        const newTodo = {
+          title: todo.fields.title,
+          id: todo.id,
+        };
+        return newTodo;
+      });
+
+      // Console.log the transformed todos array
+      console.log('Transformed Todos:', todos);
+
+      // Set the application's todoList
+      setTodoList(todos);
       setIsLoading(false);
-    });
-  }, []);
+    } catch (error) {
+      // Console.log the error's message
+      console.log(error.message);
+    }
+  };
+
+  useEffect(() => {
+  fetchData();
```

Comment:
Requires change:
This useEffect needs an empty array dependency list, otherwise, it’ll fetch data anytime React re-renders the App component due to a state change.